src/Accounts/Accounts/Account/ConnectAzureRmAccount.cs

```diff
@@ -368,7 +403,7 @@ public override void ExecuteCmdlet()
                 azureAccount.SetProperty(AzureAccount.Property.Tenants, Tenant);
             }
 
-            if (azureAccount.Type == AzureAccount.AccountType.ServicePrincipal && string.IsNullOrEmpty(CertificateThumbprint))
+            if (azureAccount.Type == AzureAccount.AccountType.ServicePrincipal && string.IsNullOrEmpty(CertificateThumbprint) && password != null)
```

Comment:
suggest to use `if (azureAccount.Type == AzureAccount.AccountType.ServicePrincipal && password != null)`